common/node_id.h

```diff
@@ -43,4 +46,21 @@ static inline int node_id_idx(const struct node_id *id1,
 /* marshal/unmarshal functions */
 void towire_node_id(u8 **pptr, const struct node_id *id);
 void fromwire_node_id(const u8 **cursor, size_t *max, struct node_id *id);
+
+/* Hash table functions for node ids */
+static inline const struct node_id *node_id_keyof(const struct node_id *id)
+{
+	return id;
+}
+
+/* We need to define a hashing function. siphash24 is a fast yet
+ * cryptographic hash in ccan/crypto/siphash24; we might be able to get away
```

Comment:
I had to dig into this a bit, it seems the hash function is collision resistant if keyed with a random key, which we do here. It is not generally cryptographically secure, and loses collision resistance if the attacker knows the key.

Reply:
Indeed!